src/rust/receiver-mock/src/metrics.rs

```diff
@@ -22,18 +37,53 @@ impl MetricsHandleResult {
     }
 }
 
+// TODO: extract this (together with internal metrics handling code to a separate file).
+
+// Would love to use predefined structs from prometheus_parse create but since those
+// don't define Serialize/Deserialize impls we can't.
+// ref: https://serde.rs/remote-derive.html
+#[derive(Debug, Deserialize, Serialize, Clone)]
+// #[derive(PartialEq)]
+pub struct Sample {
+    pub metric: String,
+    pub value: f64,
+    pub labels: HashMap<String, String>,
+    pub timestamp: i64, // milliseconds epoch timestamp
+}
+
+impl PartialEq for Sample {
+    fn eq(&self, other: &Self) -> bool {
+        self.metric == other.metric && self.labels.eq(&other.labels)
+    }
+}
+
+impl Eq for Sample {}
+
+impl Hash for Sample {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.metric.hash(state);
+
+        // Sort otherwise we get a different hash
+        let mut sorted_labels: Vec<_> = self.labels.iter().collect();
+        sorted_labels.sort_by(|x, y| x.0.cmp(&y.0));
+
+        for s in sorted_labels {
+            s.0.hash(state);
+            s.1.hash(state);
```

Comment:
I'd suggest using destructuring here, makes the code look nicer and clearer:
```rust
for (name, value) in sorted_labels {
  name.hash(state);
  value.hash(state);
}
```

Reply:
👍